ciao/arch/architecture.md

```diff
@@ -423,3 +423,88 @@ various network constructs such as subnets and interfaces are assigned names
+RBD (Rados Block Device) volumes that serve as backing images for the root file
+systems of newly created VM instances.  The image service allows users to
+create, delete, inspect and enumerate images.  Normal users can create tenant
+private images, i.e., images that are associated with a given tenant and can
+only be used to create root file systems for instances belonging to the same
+tenant.  Administrators can create public images, that that can be used by all
+users to create VM instances in any tenants they like.
+
+When a new image is uploaded to the image service it is converted into a raw
+format and stored in a new RBD volume in the ceph cluster.  An RBD snapshot is
+then taken of this new volume and the snapshot is protected.  It is this
+protected RBD snapshot that is used as the backing image for VM instances.  The
+image will be assigned a UUID when created.  It is also possible for users to
+specify a friendly name at creation time.
+
+In order to create an instance that uses an image in the image service as the
+backing image for its rootfs, the backing image must be referenced in the workload
+definition used to create the instance.  The backing image can be referenced by
+UUID or by its friendly name.  When the controller creates the new VM, it first
+creates its rootfs, by cloning the protected snapshot that represents the backing
+image.  This results in a new CIAO volume.  Volumes are discussed further in the
+following section.
+
+### Volumes
+
+The controller implements a volume service that manages a set of volumes.
+Volumes are used for two purposes:
+
+1. They serve as the root file systems of VM instances.
+2. They can be used by both VM and container instances to store data.
+
+The volume service allows volumes to be created, enumerated, inspected,
+attached and detached from existing instances.  A volume can only be
+attached to one instance at any one time.
+
+Volumes can be attached to both container and VM instances at creation time, via
+a workload definition. Typically, only newly created volumes, i.e., volumes
+created by controller when creating the instance, are attached to instances in
+this way.  It is possible to attach existing volumes to a new instance via its
+workload definition, but doing so will essentially place a restriction of one
+on the number of instances that can be created from that workload definition, as a
+volume can only be attached to one instance at any one time.
+
+Volumes created by a workload definition can be marked as ephemeral.  Ephemeral
+volumes are automatically deleted when the instance to which they are attached
+is itself deleted. Volumes automatically created to serve as the root file
+systems for instances are marked as ephemeral by default.  Non-ephemeral
+volumes, on the other hand, are not deleted along with the instances they were
+original created with.  They allow data to surive the destruction of the
```

Comment:
Maybe use deletion rather than destruction as that is the verb we use in the CLI